ambry-server/src/main/java/com/github/ambry/server/AmbryServer.java

```diff
@@ -366,16 +405,12 @@ public void startup() throws InstantiationException {
                 skipPredicate);
         replicationManager.start();
 
-
-        logger.info("Creating StatsManager to publish stats");
-
-        accountStatsMySqlStore =
-            statsConfig.enableMysqlReport ? (AccountStatsMySqlStore) new AccountStatsMySqlStoreFactory(properties,
-                clusterMapConfig, registry).getAccountStatsStore() : null;
-        statsManager =
-            new StatsManager(storageManager, clusterMap, clusterMap.getReplicaIds(nodeId), registry, statsConfig, time,
-                clusterParticipant, accountStatsMySqlStore, accountService, nodeId);
-        statsManager.start();
+        logger.info("Registering State Machine model");
+        for (ClusterParticipant participant : clusterParticipants) {
+          if (participant instanceof HelixParticipant) {
+            ((HelixParticipant) participant).registerTasksWithStateMachineModel(ambryStatsReports, accountStatsMySqlStore, accountServiceCallback);
+          }
+        }
```

Comment:
Let's expose this method in the ClusterParticipant interface, something like `startStateMachineModel`.